core/Archive/ArchivePurger.php

```diff
@@ -154,6 +156,84 @@ public function purgeOutdatedArchives(Date $dateStart)
+
+        $deletedRowCount = 0;
+        if (!empty($idArchivesToDelete)) {
+            $deletedRowCount = $this->deleteArchiveIds($dateStart, $idArchivesToDelete);
+
+            $this->logger->info("Deleted {count} rows in archive tables (numeric + blob) for deleted sites for {date}.", array(
+                'count' => $deletedRowCount,
+                'date' => $dateStart
+            ));
+
+            $this->logger->debug("[Deleted IDs: {deletedIds}]", array(
+                'deletedIds' => implode(',', $idArchivesToDelete)
+            ));
+        } else {
+            $this->logger->debug("No archives for deleted sites found in archive numeric table for {date}.", array('date' => $dateStart));
+        }
+
+        return $deletedRowCount;
+    }
+
+    public function purgeDeletedSegmentArchives(Date $dateStart, array $validSegmentHashes)
+    {
+        $idArchivesToDelete = $this->getDeletedSegmentArchives($dateStart, $validSegmentHashes);
+
+        $deletedRowCount = 0;
+        if (!empty($idArchivesToDelete)) {
+            $deletedRowCount = $this->deleteArchiveIds($dateStart, $idArchivesToDelete);
+
+            $this->logger->info("Deleted {count} rows in archive tables (numeric + blob) for deleted segments for {date}.", array(
+                'count' => $deletedRowCount,
+                'date' => $dateStart
+            ));
+
+            $this->logger->debug("[Deleted IDs: {deletedIds}]", array(
+                'deletedIds' => implode(',', $idArchivesToDelete)
+            ));
+        } else {
+            $this->logger->debug("No archives for deleted segments found in archive numeric table for {date}.", array('date' => $dateStart));
+        }
+
+        return $deletedRowCount;
+    }
+
+    protected function getDeletedSiteArchiveIds(Date $date)
+    {
+        $archiveTable = ArchiveTableCreator::getNumericTable($date);
+        return $this->model->getArchiveIdsForDeletedSites(
+            $archiveTable, 
+            $this->getOldestTemporaryArchiveToKeepThreshold()
+        );
+    }
+
+    protected function getDeletedSegmentArchives(Date $date, array $validSegmentIds)
+    {
+        $archiveTable = ArchiveTableCreator::getNumericTable($date);
+        return $this->model->getArchiveIdsForDeletedSegments(
+            $archiveTable, $validSegmentIds, $this->getOldestTemporaryArchiveToKeepThreshold()
+        );
+    }
+
+    protected function isForValidSegment($archiveRow, array $validSegmentIds)
+    {
+        $segmentHash = substr($archiveRow['name'], 4);
+
+        //Strip off the plugin name if there is one
+        $dotPlace = strpos($segmentHash, '.');
+        if ($dotPlace !== false) {
+            $segmentHash = substr($segmentHash, 0, $dotPlace);
+        }
+
+        //Check that it's in the list of segments, either for ALL sites or for THIS site
+        return  isset($validSegmentIds[0][$segmentHash]) ||
+                isset($validSegmentIds[$archiveRow['idsite']][$segmentHash]);
+    }
```

Comment:
This method doesn't appear to be used, is it still needed?

Reply:
Nope, removed